### ISSUE TITLE
False negative for rule MA0115

### ISSUE THREAD

**Issue by Fabster1993** [meziantou/Meziantou.Analyzer#731]
### Version of the Meziantou.Analyzer NuGet package

2.0.157

### Rule Identifier

MA0115

### Target Framework

.net8.0

### C# Language version

12

### Description

I am using MudBlazor combined with Meziantou.Analyzer for unknown component parameter recognition. I have a case, where Meziantou shows a false negative when using a MudBlazor component with an existing component parameter.

The parameter definition:
```c#
[Parameter]
public override InputMode InputMode { get; set; } = InputMode.numeric;
```

### Reproduction Steps

Minimal code:

```c#
<MudNumericField T="decimal" Value="Value" ValueChanged="ValueChanged" Inputmode="@(InputMode.@decimal)" />
```

**Comment by Fabster1993** [comment]
@meziantou You're right. Sorry i missed that completely! 
However, this is still a valid component parameter, since blazor does not check casing for parameters. So the code works as expected, but the analyzer shows an error...
I am not very opinionated how the behaviour of the analyzer should be in this case, so feel free to close the ticket.

Thanks anyway for the great work!

**Comment by meziantou** [comment]
I prefer keeping the existing behavior (case-sensitive). For instance, VS doesn't highlight parameters if the case doesn't match. I've improved the diagnostic message to include the list of valid parameters. It should be easier to understand the issue.
